template/bootstrap.py.jinja

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# This script initialize your repository and development environment.
+# You can re-execute it at any time
+
+import json
+import os
+from pathlib import Path
+from shutil import copy, which
+from subprocess import CalledProcessError, PIPE, check_call, check_output, run
+
+
+NOT_GIT_REPO = "fatal: not a git repository (or any of the parent directories): .git"
+ENVIRONMENT_NAME = os.environ.get("CONDA_ENVIRONMENT_NAME", "{{ python_name }}_dev")
+ENVIRONMENT_FILE = os.environ.get("CONDA_ENVIRONMENT_FILE", "environment.yml")
+
+
+def initialize_git_repository():
+    """Initialize git repository"""
+    git_status = run(["git", "status"], stdout=PIPE, stderr=PIPE, encoding="utf-8")
+    if git_status.returncode > 0 and git_status.stderr.strip() == NOT_GIT_REPO:
+        print("\nInitializing git...")
+        try:
+            check_call(["git", "init"])
+            check_call(["git", "branch", "-m", "main"])
+        except CalledProcessError as e:
+            print(f"Failed to initialize git repository:\n{e!s}\n")
+        else:
+            print("Git initialized for the extension\n")
+
+    # Copy pre-commit hook
+    copy("scripts/pre-commit", ".git/hooks")
```

Comment:
Why not use `pre-commit`?

Reply:
Thinking that if someone runs this bootstrap script but then would like to disable the pre-commit hook they might not know where it came from.

Reply:
> Why not use `pre-commit`?

I want only to reuse scripts we have. I am not sure about bringing pre-commit as it will add yet another tool (of which I'm not a fan).

> Thinking that if someone runs this bootstrap script but then would like to disable the pre-commit hook they might not know where it came from.

I certainly can document it more and add some stdout.

Reply:
Or maybe we remove the `pre-commit` script / hook, for simplicity?

The CI workflow should catch lint issues anyway.